src/hal/ticks.h

```diff
@@ -0,0 +1,14 @@
+/*
+ * SPDX-License-Identifier: MIT
+ * SPDX-FileCopyrightText: 2021 Rafael Silva <silvagracarafael@gmail.com>
+ */
+
+#pragma once
+
+#include "util/types.h"
+
+struct ticks_hal_t {
+	u64 (*ticks_ms)();
+	void (*delay_ms)(u32 ticks);
+	void (*delay_us)(u32 ticks);
```

Comment:
Let's add `drv_data` here, we don't need it for the STM32F1 but we might for other platforms. As a general rule, all HALs should have a `drv_data` pointer, so that platforms can use it if they need it.

Reply:
sure